docs/pf-keys/parflow/keys_contribution.rst

```diff
@@ -0,0 +1,175 @@
+to the file (e.g. ``Solver.CLM.MetFile``). If no arguments are provided, it will check your current working directory for the file.
+
+
+^^^^^^^^^^^^^^^^^
+``handlers``
+^^^^^^^^^^^^^^^^^
+
+This will help generate dynamically defined tokens within other keys based on the provided value for the key. Each
+argument is an updater that specifies where and how the value is used to create other tokens. An example from phase.yaml
+is below:
+
+.. code-block:: python3
+
+            Phase:
+                Names:
+                    handlers:
+                        PhaseUpdater:
+                            type: ChildrenHandler
+                            class_name: PhaseNameItem
+                            location: .
+
+``PhaseUpdater`` is the name of the handler. The arguments for the handler include ``type``, ``class_name``, and ``location``.
+The only current option for ``type`` is ``ChildrenHandler``. ``class_name`` corresponds to the ``__class__`` annotation of the
+dynamic token. In this example, ``PhaseNameItem`` is the ``__class__`` of the dynamic token ``.{phase_name}``. ``location`` is
+the location of the token referenced in ``class_name``. In this example, the Names token in ``Phase.Names`` is on the same
+level as the ``.{phase_name}`` in ``Phase.phase_name``. This can also be an absolute path.
+
+
+===================================================
+Steps to add a new key
+===================================================
+
+1. Select the yaml file that most closely matches the key that you want to add. If your key is a token nested within an
+existing key, be sure to find which yaml file includes the parent token(s). For example, if you wanted to add the key
+``Solver.Linear.NewKey``, you would add it within the file *solver.yaml*.
+
+2. Open the yaml file and navigate to the level within the hierarchy where you want to put your key. The structure of
+the yaml files is designed to be easy to follow, so it should be easy to find the level where you'd like to add your
+key. The indentation of these files is two spaces. Using our ``Solver.Linear.NewKey`` example, ``Solver`` is at the far
+left, ``Linear`` is two spaces (one tab) in, and you would add ``NewKey`` two more spaces in (two tabs). We suggest copying
+and pasting an existing key from the same level to make sure it's correct.
+
+3. Fill in the details of your key. Again, this format is designed to be readable, so please refer to examples in the
+yaml files to guide you. The details you can include are listed in the section above.
+
+4. Regenerate the Python keys using ``make GeneratePythonKeys``.
+
+You should see a longer message indicating an update that lists the overlapping classes, including the line ``Defined ##
```

Comment:
The class name that get generated are coming from the PF Key name and as you can imagine, there are conflict (i.e. Alpha) so we allow the user to come up with their own name to prevent conflict. 

We generate that many classes to allow IDE to have code completion with inline documentation.